fields/sql_test.go

```diff
@@ -131,53 +131,69 @@ func TestField_Value(t *testing.T) {
+		{Type: int32(5), Out: int32(5), In: int64(5)},
+		{Type: likeInt(5), Out: likeInt(5), In: int64(5)},
+		{Type: float32(5), Out: float32(5), In: float64(5)},
+		{Type: likeFloat(5), Out: likeFloat(5), In: float64(5)},
 		// Interfaces without tags:
-		{Out: ifaceScanner{[]byte("scan me")}, In: []byte("scan me")},
-		{Out: ifaceValuer{}, In: []byte("value"), Error: true},
-		{Out: ifaceMarshal{}, In: ifaceMarshal{}, Error: true},
-		{Out: ifaceBinaryMarshal{}, In: ifaceBinaryMarshal{}, Error: true},
-		{Out: ifaceTextMarshal{}, In: ifaceTextMarshal{}, Error: true},
-		{Out: ifaceJSONMarshal{}, In: ifaceJSONMarshal{}, Error: true},
+		{Type: ifaceScanner{[]byte("scan me")}, Out: ifaceScanner{[]byte("scan me")}, In: []byte("scan me")},
+		{Type: ifaceMarshal{}, Out: ifaceMarshal{}, In: []byte{}, Error: true},
+		{Type: ifaceBinaryMarshal{}, Out: ifaceBinaryMarshal{}, In: []byte{}, Error: true},
+		{Type: ifaceTextMarshal{}, Out: ifaceTextMarshal{}, In: []byte{}, Error: true},
+		{Type: ifaceJSONMarshal{}, Out: ifaceJSONMarshal{}, In: []byte{}, Error: true},
+		// Pointer scanner with nil:
+		{Type: &ifaceScanner{}, Out: (*ifaceScanner)(nil), In: nil},
+		// Pointer scanner with value:
+		{Type: &ifaceScanner{}, Out: &ifaceScanner{[]byte("scan me")}, In: []byte("scan me")},
 		// Interfaces with tags:
-		{Out: ifaceMarshal{"binary_one"}, In: []byte("binary_one"), Tag: "binary"},
-		{Out: ifaceBinaryMarshal{"binary_two"}, In: []byte("binary_two"), Tag: "binary"},
-		{Out: ifaceTextMarshal{"text"}, In: []byte("text"), Tag: "string"},
-		{Out: ifaceJSONMarshal{[]string{"json"}}, In: []byte("[\"json\"]"), Tag: "json"},
+		{Type: ifaceMarshal{"binary_one"}, Out: ifaceMarshal{"binary_one"}, In: []byte("binary_one"), Tag: "binary"},
+		{Type: ifaceBinaryMarshal{"binary_two"}, Out: ifaceBinaryMarshal{"binary_two"}, In: []byte("binary_two"), Tag: "binary"},
+		{Type: ifaceTextMarshal{"text"}, Out: ifaceTextMarshal{"text"}, In: []byte("text"), Tag: "string"},
+		{Type: ifaceJSONMarshal{[]string{"json"}}, Out: ifaceJSONMarshal{[]string{"json"}}, In: []byte("[\"json\"]"), Tag: "json"},
+		// Pointer interfaces with tags with nil:
+		{Type: &ifaceMarshal{}, Out: (*ifaceMarshal)(nil), In: nil, Tag: "binary"},
+		{Type: &ifaceBinaryMarshal{}, Out: (*ifaceBinaryMarshal)(nil), In: nil, Tag: "binary"},
+		{Type: &ifaceTextMarshal{}, Out: (*ifaceTextMarshal)(nil), In: nil, Tag: "string"},
+		{Type: &ifaceJSONMarshal{}, Out: (*ifaceJSONMarshal)(nil), In: nil, Tag: "json"},
+		// Pointer interfaces with tags with value:
+		{Type: &ifaceMarshal{}, Out: &ifaceMarshal{"binary_one"}, In: []byte("binary_one"), Tag: "binary"},
+		{Type: &ifaceBinaryMarshal{}, Out: &ifaceBinaryMarshal{"binary_two"}, In: []byte("binary_two"), Tag: "binary"},
+		{Type: &ifaceTextMarshal{}, Out: &ifaceTextMarshal{"text"}, In: []byte("text"), Tag: "string"},
+		{Type: &ifaceJSONMarshal{}, Out: &ifaceJSONMarshal{[]string{"json"}}, In: []byte("[\"json\"]"), Tag: "json"},
```

Comment:
curious to see what happens if for some bizarre reason someone defines field type to be `Type: interface{}(nil)`